### DISCUSSION TITLE
Should conversation mode be enabled by default?

### DISCUSSION THREAD

**Discussion by ngxson** [ggml-org/llama.cpp#8303] [General]
Maybe a controversy topic to discuss, but seeing other CLI apps like [local-gemma](https://github.com/huggingface/local-gemma) or [ollama](https://github.com/ollama/ollama) starts in chat mode (conversation mode) by default, does it make sense for `llama-cli` to do the same thing?

Support for chat template is also much better now. Most of the time, simply adding `-cnv` without further tweaking is already enough.

The advantage of this change is that it is more intuitive for most users. Chat has became the "status quo" these days when running LLMs.

The disadvantage is that this can be a breaking change if users already use `llama-cli` in an automated way (I remember someone use `llama-cli` in their shell script to generate commands)

**Comment by ggerganov** [comment]
Don't think it's worth it - the `-cnv` flag is very simple to use. Also, I almost never run chat mode because I debug typically with base models - maybe other developers do the same.

**Comment by dspasyuk** [comment]
@ngxson I think it would be the best course of action too, thank you for your help BTW. It is fairly easy to use but it is not even documented in the manual at the bottom of the README page. I think debugging should be done in that mode too. Is there a reason to load/unload the model into memory every single time? Does it provide any benefit to -cnv mode?

**Comment by danielzgtg** [comment]
This was implemented in #11214
